tests/pipelines/pixart/test_pixart.py

```diff
@@ -410,3 +410,22 @@ def test_pixart_1024_without_resolution_binning(self):
         no_res_bin_image_slice = no_res_bin_image[0, -3:, -3:, -1]
 
         assert not np.allclose(image_slice, no_res_bin_image_slice, atol=1e-4, rtol=1e-4)
+
+    def test_pixart_512_without_resolution_binning(self):
+        generator = torch.manual_seed(0)
+
+        pipe = PixArtAlphaPipeline.from_pretrained("PixArt-alpha/PixArt-XL-2-512x512", torch_dtype=torch.float16)
+        pipe.enable_model_cpu_offload()
+
+        prompt = "A small cactus with a happy face in the Sahara desert."
+
+        image = pipe(prompt, generator=generator, num_inference_steps=5, output_type="np").images
+        image_slice = image[0, -3:, -3:, -1]
+
+        generator = torch.manual_seed(0)
+        no_res_bin_image = pipe(
+            prompt, generator=generator, num_inference_steps=5, output_type="np", use_resolution_binning=False
+        ).images
+        no_res_bin_image_slice = no_res_bin_image[0, -3:, -3:, -1]
+
+        assert not np.allclose(image_slice, no_res_bin_image_slice, atol=1e-4, rtol=1e-4)
```

Comment:
I don't understand this test 

I think `use_resolution_binning` only come into effect when you want to generate non square images, i.e. when `height` and `width` is not the same - here both are `None` so it should not make a difference

why do we expect the output for `use_resolution_binning = True` and `use_resolution_binning=False` to be different?

Reply:
So, it's basically testing that here, actually i.e., when resolution binning is not effective when the height == width. But I guess it's not comprehensive enough that way. I will modify it accordingly.